### PR TITLE
Same key rules for map accesses

### PR DESCRIPTION
**Pull request by ehildenb**
-   Formats single-key accesses one line.
-   Adds in many rules for same-key accesses to maps (where no update is expected).

### CODE REVIEW

File: gem.md [hunk 1]
```diff
@@ -97,6 +97,15 @@ Gem Semantics
       requires VALUE >=Rat 0
        andBool BALANCE_SRC >=Rat VALUE
```
Comment (malturki) [inline gem.md]: In dai.md above, there is also the check that the two accounts must be different, even though we are matching on two entries in the balances map. For consistency, we should probably do the same here: `ACCTSRC =/=K ACCTDST`, right?

File: gem.md [hunk 2]
```diff
@@ -135,11 +140,7 @@ Gem Semantics
...
            <gem-id> GEMID </gem-id>
-           <gem-balances>
-             ...
-             ACCTSRC |-> ( BALANCE_SRC => BALANCE_SRC -Rat VALUE )
-             ...
-           </gem-balances>
+           <gem-balances> ... ACCTSRC |-> ( BALANCE_SRC => BALANCE_SRC -Rat VALUE ) ... </gem-balances>
            ...
          </gem>
       requires VALUE >=Rat 0
```
Comment (malturki) [inline gem.md]: Shouldn't we have a check here that the amount being burned is not more than the account's balance (I haven't actually checked the contract source)?

File: gem.md (hunk 1 above)
Comment (ehildenb) [reply gem.md]: Fixed.

File: gem.md (hunk 2 above)
Comment (ehildenb) [reply gem.md]: Fixed.

Comment (ehildenb) [reply gem.md]: The `gem` contracts are third-party supplied, so this is just a model of a "generic" gem contract. So it makes sense to make it as "tight" as possible.

**Pull request merged**